google/cloud/storage/oauth2/google_credentials.h

```diff
@@ -62,21 +64,79 @@ CreateAuthorizedUserCredentialsFromJsonFilePath(std::string const&);
  * with Cloud Storage client libraries.
  */
 StatusOr<std::shared_ptr<Credentials>>
-CreateAuthorizedUserCredentialsFromJsonContents(std::string const&);
+CreateAuthorizedUserCredentialsFromJsonContents(std::string const& contents);
 
-/// Creates a ServiceAccountCredentials rom a JSON file at the specified path.
+/**
+ * Creates a ServiceAccountCredentials from a JSON file at the specified path.
+ *
+ * These credentials use the cloud-platform OAuth 2.0 scope, defined by
+ * `GoogleOAuthScopeCloudPlatform()`. To specify alternate scopes, use the
+ * overloaded version of this function.
+ */
+StatusOr<std::shared_ptr<Credentials>>
+CreateServiceAccountCredentialsFromJsonFilePath(std::string const& path);
+
+/**
+ * Creates a ServiceAccountCredentials from a JSON file at the specified path.
+ *
+ * @param path the path to the file containing service account JSON credentials.
+ * @param scopes the scopes to request during the authorization grant. If
+ *     omitted or an empty set is provided, the cloud-platform scope,
```

Comment:
`scopes` can't be omitted; it's required. If it's logically optional, perhaps it should also be wrapped in a std::optional?